cdap-app-fabric/src/main/java/io/cdap/cdap/internal/capability/CapabilityApplier.java

```diff
@@ -286,6 +312,135 @@ private void deployApp(SystemApplication application) throws Exception {
                  application.getArtifact(), configString, NOOP_PROGRAM_TERMINATOR, null, null);
   }
 
+  @VisibleForTesting
+  void autoInstallResources(String capability, List<AutoInstall> autoInstall) {
+    if (autoInstall == null || autoInstall.isEmpty()) {
+      LOG.debug("Capability {} do not have auto install resources associated with it", capability);
+      return;
+    }
+
+    for (AutoInstall a : autoInstall) {
+      Package[] packages;
+      String hubUrl = a.getHubUrl();
+      try {
+        URL url = new URL(new URL(hubUrl), "/v2/packages.json");
+        String packagesJson = doGetAsString(url);
+        // Deserialize packages.json from hub
+        // See https://cdap.atlassian.net/wiki/spaces/DOCS/pages/554401840/Hub+API?src=search#Get-Hub-Catalog
+        packages = GSON.fromJson(packagesJson, Package[].class);
+      } catch (Exception e) {
+        LOG.error("Failed to get packages.json from {}", hubUrl, e);
```

Comment:
If the exception is actually ignored, log it as warning and stating clearly that it is being ignored.

Reply:
fixed